### PR TITLE
Fix Gemma4 QAT loading and cache defaults for Osaurus

### PR DESCRIPTION
**Pull request by jjang-ai**
## Summary

Gemma 4 QAT correctness checkpoint for Osaurus.

This PR fixes the vMLX-side runtime/load/parser/cache pieces needed by Osaurus PR #1469 for Gemma 4 QAT MXFP4 and JANG_4M rows:

- loads Gemma 4 QAT per-layer projection tensors with the correct quantized shape context
- accepts `gemma4_unified` / `gemma4_unified_text` aliases where needed
- tolerates unified config gaps such as missing `vision_soft_tokens_per_image`
- skips unsupported `vision_embedder.*` keys only for text-only loads
- remaps fused MoE `gate_up_proj`/`down_proj` tensors into the expected SwitchGLU expert layout
- preserves Gemma parser/tool-call fixes for native and observed Zyphra/Gemma drift formats without visible protocol leakage
- defaults server/runtime cache settings so ordinary single-batch loading does not enable paged RAM KV cache
- keeps engine-selected KV codec, block-disk/L2 cache settings, and prefill progress reporting available to Osaurus
- adds Model2Vec static embedding APIs required by current Osaurus main

## Current Scope

This is the vMLX dependency for the Osaurus Gemma correctness PR. Raw speed work and Gemma4 audio are intentionally deferred. Do not use BF16/source Gemma bundles as this checkpoint's load target.

Cache claims must follow telemetry. If a runtime row reports rotating KV plus disk-backed restore and `turbo_quant_kv_layer_count=0`, the row should be documented that way rather than promoted as a TurboQuant-KV-layer topology.

## Validation

Local proof after merging current vMLX main:

- `git diff --check`
- `DEVELOPER_DIR=/Applications/Xcode.app/Contents/Developer swift build --target MLXLMCommon --target MLXLLM --target MLXVLM`
  - log: `/tmp/vmlx-gemma4-qatsidecar-build-20260612T202403Z/build-mlxllm-mlxvlm-common.log`
- Focused source-contract test run covered server settings, Gemma parser/no-leak contracts, cache topology contracts, prefill prep, and BatchEngine TurboQuant contracts before hitting this machine's local MLX `default.metallib` discovery boundary:
  - log: `/tmp/vmlx-gemma4-qatsidecar-main-merge-20260612T201533Z/focused-swift-tests.log`

The local Swift test runner path is not a clean full-runtime proof on this machine right now because reruns can hang after MLX `default.metallib` initialization. The Osaurus PR carries the app-facing proof and CI gate for the consumed pin.

## Osaurus Dependency

Osaurus PR #1469 is green and mergeable against this vMLX branch head. After this vMLX PR lands on main, Osaurus must repin to the resulting vMLX main SHA if it differs from this PR head.

### CODE REVIEW

**Comment by jjang-ai** [comment]
Speed attribution checkpoint for Gemma 4 QAT MXFP4/JANG_4M.

Changes in this checkpoint:
- Replaced the custom Gemma4 PLE scaled-linear modules with normal `Linear` plus post-projection scale in both text and VLM paths, matching the upstream quantizable PLE design.
- Kept the VLM router parity fix: scaled RMSNorm, top-k over raw logits, softmax over selected logits only.
- Updated `docs/GEMMA4_QAT_SPEED_METHOD_2026_06_12.md` with the current matrix, GGUF gap, graph stats, compiled-decode diagnostic, TurboQuant KV diagnostic, rejected PLE dequant row, and next bottleneck.

Fresh direct RunBench release results, no Osaurus/UI timing:

| Model | Quant | Deterministic | Bundle defaults | Caveat |
|---|---|---:|---:|---|
| E2B | MXFP4 | 146.7 tok/s | 136.9 tok/s | clean speed row |
| E2B | JANG_4M | 135.2 tok/s | 127.7 tok/s | clean speed row |
| E4B | MXFP4 | 93.8 tok/s | 90.4 tok/s | clean speed row |
| E4B | JANG_4M | 82.6 tok/s | 80.4 tok/s | clean speed row |
| 12B | MXFP4 | 39.8 tok/s | 37.8 tok/s | deterministic loop=YES |
| 12B | JANG_4M | 32.7 tok/s | 32.6 tok/s | clean speed row |
| 26B A4B | MXFP4 | 97.2 tok/s | 91.9 tok/s | bundle unclosedReasoning=YES |
| 26B A4B | JANG_4M | 84.2 tok/s | 81.1 tok/s | clean speed row |
| 31B | MXFP4 | 18.4 tok/s | 18.8 tok/s | clean speed row |
| 31B | JANG_4M | 17.2 tok/s | 16.6 tok/s | bundle unclosedReasoning=YES |

E2B GGUF baseline is 173.677 tok/s decode from `/tmp/vmlx-gemma4-e2b-compare-20260612T155643Z/gguf-llama-bench.json`.
- E2B MXFP4 is 15.5% slower than GGUF.
- E2B JANG_4M is 22.2% slower than GGUF.

Diagnostics:
- Compiled batch decode helps dense rows but does not close parity: 12B MXFP4 39.8 -> 50.4 tok/s, 31B MXFP4 18.4 -> 23.4 tok/s. 12B still has loop=YES.
- TurboQuant KV is not the single-batch short-context speed fix: E2B 146.7 -> 89.7 tok/s, 12B 39.8 -> 32.5 tok/s, 31B 18.4 -> 16.5 tok/s.
- PLE dense dequantization was tested and rejected as slower.
- Current root speed target is dense Gemma4 quantized matmul/attention scaling plus compiled decode promotion/correctness, not Osaurus request overhead.

Verification:
- `swift build -c release --product RunBench --jobs 4` passed.
- Full direct speed artifacts are recorded in the doc.
- `git diff --check` passed before commit.
- Broad `swift test --list-tests` is still blocked by the known shell toolchain `no such module 'Testing'` issue.
- `scripts/vmlx-architecture-cache-proof-check.sh` is not green on this branch due older broad proof-text expectations and a stale macOS CodeSigningHelper XPC process; not counted as a Gemma speed pass.
